### PR TITLE
Switch test frameworks to mocha, expect and istanbul

### PR DESCRIPTION
**Pull request by sttk**
I modified test codes for **mocha** + **expect**.
And I made enable to measure their coverage with **istanbul**.

### CODE REVIEW

**Comment by sttk** [comment]
Sorry. I forgot to modify `appveyor.yml` for Windows.

File: package.json [hunk 1]
```diff
@@ -23,11 +23,12 @@
     "gulp.1"
   ],
   "scripts": {
-    "coveralls": "lab -r lcov | coveralls",
     "lint": "eslint . && jscs index.js bin/ lib/ test/",
     "prepublish": "marked-man --name gulp docs/CLI.md > gulp.1",
     "pretest": "npm run lint",
-    "test": "lab test/*.js -cv -I Reflect",
+    "test": "mocha --reporter spec",
```
Comment (phated) [inline package.json]: I thought `spec` was the default reporter.  You can use the same set of scripts as https://github.com/gulpjs/vinyl/blob/master/package.json#L22-L26

File: test/completion.js [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
   ['bash', 'fish', 'powershell', 'zsh'].forEach(function(type) {
-    lab.test('returns completion script for ' + type, function(done) {
+    it('returns completion script for ' + type, function(done) {
       runner({ verbose: false })
         .gulp('--completion=' + type)
         .run(cb);
 
       function cb(err, stdout) {
-        expect(stdout).to.contain('gulp --completion=' + type);
+        expect(stdout).toMatch('gulp --completion=' + type);
```
Comment (phated) [inline test/completion.js]: why `toMatch` instead of `toEqual`?

File: test/completion.js [hunk 3]
```diff
@@ -1,31 +1,30 @@
...
 
-  lab.test('shows error message for unknown completion type', function(done) {
+  it('shows error message for unknown completion type', function(done) {
     runner({ verbose: false })
       .gulp('--completion=unknown')
       .run(cb);
 
     function cb(err, stdout) {
-      expect(stdout).to.contain('rules for \'unknown\' not found');
+      expect(stdout).toMatch('rules for \'unknown\' not found');
```
Comment (phated) [inline test/completion.js]: why `toMatch` instead of `toEqual`?

File: test/flags-continue.js [hunk 4]
```diff
@@ -1,25 +1,24 @@
...
 
-  lab.test('continues execution when flag is set', function(done) {
+  it('continues execution when flag is set', function(done) {
     runner({ verbose: false })
       .gulp('test4', '--continue', '--cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
     function cb(err, stdout, stderr) {
-      expect(err).to.be.not.null();
+      expect(err).toNotBe(null);
```
Comment (phated) [inline test/flags-continue.js]: why `toNotBe` instead of `toNotEqual`?

File: test/flags-continue.js [hunk 5]
```diff
@@ -28,32 +27,32 @@ lab.experiment('flag: --continue', function() {
...
 
-  lab.test('stops execution when flag is not set', function(done) {
+  it('stops execution when flag is not set', function(done) {
     runner({ verbose: false })
       .gulp('test4', '--cwd ./test/fixtures/gulpfiles')
       .run(cb);
 
     function cb(err, stdout, stderr) {
-      expect(err).to.be.not.null();
+      expect(err).toNotBe(null);
```
Comment (phated) [inline test/flags-continue.js]: why `toNotBe` instead of `toNotEqual`?

File: test/flags-verify.js [hunk 6]
```diff
@@ -1,23 +1,22 @@
...
 
-  lab.test('dependencies with invalid dependency', function(done) {
+  it('dependencies with invalid dependency', function(done) {
     runner({ verbose: false })
       .gulp('--verify invalid-package.json', '--cwd ./test/fixtures/packages/')
       .run(cb);
 
     function cb(err, stdout) {
-      expect(err).to.be.not.null();
+      expect(err).toNotBe(null);
```
Comment (phated) [inline test/flags-verify.js]: why `toNotBe` instead of `toNotEqual`?

File: test/flags-verify.js [hunk 7]
```diff
@@ -47,16 +46,16 @@ lab.experiment('flag: --verify', function() {
...
 
-  lab.test('default args with invalid dependency', function(done) {
+  it('default args with invalid dependency', function(done) {
     runner({ verbose: false })
       .gulp('--verify', '--cwd ./test/fixtures/packages/')
       .run(cb);
 
     function cb(err, stdout) {
-      expect(err).to.be.not.null();
+      expect(err).toNotBe(null);
```
Comment (phated) [inline test/flags-verify.js]: why `toNotBe` instead of `toNotEqual`?

File: test/logging.js [hunk 8]
```diff
@@ -1,54 +1,82 @@
...
 
-lab.experiment('logging', function() {
-  lab.test('log-level flag for debug: -LLLL', function(done) {
-    child.exec('node ' + __dirname + '/fixtures/logging.js -LLLL', function(err, stdout, stderr) {
+// jscs:disable requireCamelCaseOrUpperCaseIdentifiers
+var isIstanbul = process.env.running_under_istanbul;
+// jscs:enable
+
+var cmd;
+if (isIstanbul) {
```
Comment (phated) [inline test/logging.js]: what is this doing?

**Comment by phated** [comment]
@sttk an aside: do you think we should be using https://www.npmjs.com/package/nyc for istanbul coverage because we spawn child processes?

**Comment by sttk** [comment]
I modified what you pointed out, and made use of **nyc** for coverage.

**Comment by phated** [comment]
@sttk great work once again!  Thank you so much!!!

**Comment by sttk** [comment]
@phated Thank you for reviewing and merging this PR.

**Pull request merged**